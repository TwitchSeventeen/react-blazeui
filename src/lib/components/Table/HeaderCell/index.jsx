import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function HeaderCell(props) {
  const {
    children, className, ...other
  } = props;

  const classes = classNames(
    'c-table__cell',
    className,
  );

  return (
    <th className={classes} {...other}>
      { children }
    </th>
  );
}

HeaderCell.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

HeaderCell.defaultProps = {
  className: null,
  children: null,
};

export default HeaderCell;
