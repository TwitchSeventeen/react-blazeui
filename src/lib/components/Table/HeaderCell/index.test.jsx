import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HeaderCell from './index';

describe('HeaderCell', () => {
  it('should be defined', () => {
    expect(HeaderCell).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <HeaderCell>
        lorem ipsum dolor
      </HeaderCell>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
