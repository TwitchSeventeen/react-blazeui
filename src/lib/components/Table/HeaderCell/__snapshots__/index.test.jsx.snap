// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCell renders correctly 1`] = `
<th
  className="c-table__cell"
>
  lorem ipsum dolor
</th>
`;
